engine/engine_test.go

```diff
@@ -786,6 +786,78 @@ func TestQueriesAgainstOldEngine(t *testing.T) {
 	+ on() group_left()
 	sum(http_requests_total{ns="nginx"})`,
 		},
+		// Result is correct but this likely fails due to https://github.com/golang/go/issues/12025.
+		// TODO(saswatamcode): Test NaN cases separately. https://github.com/thanos-community/promql-engine/issues/88
+		// {
+		// 	name: "scalar func with NaN",
+		// 	load: `load 30s
+		//  	http_requests_total{pod="nginx-1"} 1+1x15
+		//  	http_requests_total{pod="nginx-2"} 1+2x18`,
+		// 	query: `scalar(http_requests_total)`,
+		// },
+		{
+			name: "scalar func with aggr",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `scalar(max(http_requests_total))`,
+		},
+		{
+			name: "scalar func with number",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `scalar(max(http_requests_total)) + 10`,
+		},
+		{
+			name: "clamp",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp(http_requests_total, 5, 10)`,
+		},
+		{
+			name: "clamp_min",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp_min(http_requests_total, 10)`,
+		},
+		{
+			name: "complex func query",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp(1 - http_requests_total, 10 - 5, 10)`,
+		},
+		{
+			name: "func within func query",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp(irate(http_requests_total[30s]), 10 - 5, 10)`,
+		},
+		{
+			name: "aggr within func query",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp(rate(http_requests_total[30s]), 10 - 5, 10)`,
+		},
+		{
+			name: "func with scalar arg that selects storage",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp_min(http_requests_total, scalar(max(http_requests_total)))`,
+		},
+		{
+			name: "func with scalar arg that selects storage + number",
+			load: `load 30s
+			http_requests_total{pod="nginx-1"} 1+1x15
+			http_requests_total{pod="nginx-2"} 1+2x18`,
+			query: `clamp_min(http_requests_total, scalar(max(http_requests_total)) + 10)`,
```

Comment:
Can we update the readme with the new benchmark result?

Reply:
Sure will do!

Reply:
Updated the single-core one. For multi-core I don't have an 8-core machine 😅